wandb/apis/importers/base.py

```diff
@@ -0,0 +1,309 @@
+        return {}
+
+    def summary(self) -> Dict[str, float]:
+        return {}
+
+    def metrics(self) -> List[Dict[str, float]]:
+        """
+        We expect metrics in this shape:
+        [
+            {'metric1': 1, 'metric2': 1, '_step': 0},
+            {'metric1': 2, 'metric2': 4, '_step': 1},
+            {'metric1': 3, 'metric2': 9, '_step': 2},
+            ...
+        ]
+
+        You can also submit metrics in this shape:
+        [
+            {'metric1': 1, '_step': 0},
+            {'metric2': 1, '_step': 0},
+            {'metric1': 2, '_step': 1},
+            {'metric2': 4, '_step': 1},
+            ...
+        ]
+        """
+        return []
+
+    def run_group(self) -> Optional[str]:
+        ...
+
+    def job_type(self) -> Optional[str]:
+        ...
+
+    def display_name(self) -> str:
+        return self.run_id()
+
+    def notes(self) -> Optional[str]:
+        ...
+
+    def tags(self) -> Optional[List[str]]:
+        ...
+
+    def artifacts(self) -> Optional[Iterable[Tuple[Name, Path]]]:
+        ...
+
+    def os_version(self) -> Optional[str]:
+        ...
+
+    def python_version(self) -> Optional[str]:
+        ...
+
+    def cuda_version(self) -> Optional[str]:
+        ...
+
+    def program(self) -> Optional[str]:
+        ...
+
+    def host(self) -> Optional[str]:
+        ...
+
+    def username(self) -> Optional[str]:
+        ...
+
+    def executable(self) -> Optional[str]:
+        ...
+
+    def gpus_used(self) -> Optional[str]:
+        ...
+
+    def cpus_used(self) -> Optional[int]:  # can we get the model?
+        ...
+
+    def memory_used(self) -> Optional[int]:
+        ...
+
+    def runtime(self) -> Optional[int]:
+        ...
+
+    def start_time(self) -> Optional[int]:
+        ...
+
+    def make_run_record(self) -> pb.Record:
+        run = pb.RunRecord()
+        run.run_id = self.run_id()
+        run.entity = self.entity()
+        run.project = self.project()
+        run.display_name = coalesce(self.display_name())
+        run.notes = coalesce(self.notes(), "")
+        run.tags.extend(coalesce(self.tags(), list()))
+        # run.start_time.FromMilliseconds(self.start_time())
+        # run.runtime = self.runtime()
+        run_group = self.run_group()
+        if run_group is not None:
+            run.run_group = run_group
+        self.interface._make_config(
+            data=self.config(),
+            obj=run.config,
+        )  # is there a better way?
+        return self.interface._make_record(run=run)
+
+    def make_summary_record(self) -> pb.Record:
+        d: dict = {
+            **self.summary(),
+            "_runtime": self.runtime(),  # quirk of runtime -- it has to be here!
+            # '_timestamp': self.start_time()/1000,
+        }
+        summary = self.interface._make_summary_from_dict(d)
+        return self.interface._make_record(summary=summary)
+
+    def make_history_records(self) -> Iterable[pb.Record]:
```

Comment:
Is there a reason for these to be exported? Seems like these should be implementation details.

Reply:
agreed... you might want to underscore the things that you dont expect the user to interact with.